### PR TITLE
🐛 Fix okta incremental stream

### LINKED ISSUE #5828: Okta Pagination exception
## Current Behavior

From [Slack](https://airbytehq.slack.com/archives/C01MFR03D5W/p1630634255318300)

Hello, when using Okta source with Incremental sync, it fails to fetch the details. Reason is url being queried is of the form https://xxxx.okta.com/api/v1/users?limit=100&filter=lastUpdated+gt+2021-07-01T17%3A05%3A06.000Z while the correct url should be of type https://xxxx.okta.com/api/v1/users?limit=100&filter=lastUpdated+gt+"2021-07-01T17%3A05%3A06.000Z", i.e. with double quotes around the time being specified

## Acceptance Criteria
1. reproduce the error
2. create a unit test
3. fix the error

### PR DESCRIPTION
**Pull request by avida**
## What

Resolves #5828 

## Recommended reading order
1. `x.java`
2. `y.python`

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector added to connector index like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-okta/acceptance-test-config.yml [hunk 1]
```diff
@@ -15,3 +15,9 @@ tests:
   full_refresh:
     - config_path: "secrets/config.json"
       configured_catalog_path: "integration_tests/configured_catalog.json"
+  incremental:
+    - config_path: "secrets/config.json"
+      configured_catalog_path: "integration_tests/configured_catalog_incremental.json"
+      future_state_path: "integration_tests/abnormal_state.json"
+      cursor_paths:
+        charges: ["lastUpdated"]
```
Comment (Zirochkaa) [inline airbyte-integrations/connectors/source-okta/acceptance-test-config.yml:23]: Maybe `users`?
```suggestion
        users: ["lastUpdated"]
```

File: airbyte-integrations/connectors/source-okta/integration_tests/configured_catalog_incremental.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
+  "streams": [
+    {
+      "stream": {
+        "name": "users",
+        "json_schema": {}
```
Comment (Zirochkaa) [inline airbyte-integrations/connectors/source-okta/integration_tests/configured_catalog_incremental.json:6]: What about `supported_sync_modes`?

File: airbyte-integrations/connectors/source-okta/integration_tests/configured_catalog_incremental.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{
+  "streams": [
+    {
+      "stream": {
+        "name": "users",
```
Comment (Zirochkaa) [inline airbyte-integrations/connectors/source-okta/integration_tests/configured_catalog_incremental.json:5]: Why do you have `users` here and also in `configured_catalog.json` file? Please merge these files into one and use this one file in tests.

**Review by Zirochkaa — CHANGES_REQUESTED** [review]
Also, what about incremental tests for `groups` stream? Maybe we can add them too?

File: airbyte-integrations/connectors/source-okta/integration_tests/configured_catalog_incremental.json (hunk 3 above)
Comment (avida) [reply airbyte-integrations/connectors/source-okta/integration_tests/configured_catalog_incremental.json]: Done

File: airbyte-integrations/connectors/source-okta/integration_tests/configured_catalog_incremental.json (hunk 2 above)
Comment (avida) [reply airbyte-integrations/connectors/source-okta/integration_tests/configured_catalog_incremental.json]: Updated

File: airbyte-integrations/connectors/source-okta/acceptance-test-config.yml (hunk 1 above)
Comment (avida) [reply airbyte-integrations/connectors/source-okta/acceptance-test-config.yml]: Done

File: airbyte-integrations/connectors/source-okta/integration_tests/configured_catalog.json [hunk 4]
```diff
@@ -4,7 +4,8 @@
       "stream": {
         "name": "users",
         "json_schema": {},
-        "supported_sync_modes": ["full_refresh"]
+        "supported_sync_modes": ["full_refresh"],
+        "supported_sync_modes": ["full_refresh", "incremental"]
```
Comment (Zirochkaa) [inline airbyte-integrations/connectors/source-okta/integration_tests/configured_catalog.json:8]: Now you have 2 `supported_sync_modes` :)

**Review by Zirochkaa — APPROVED** [review]
lgtm, just a small comment.

File: airbyte-integrations/connectors/source-okta/integration_tests/configured_catalog.json (hunk 4 above)
Comment (avida) [reply airbyte-integrations/connectors/source-okta/integration_tests/configured_catalog.json:7]: updated

**Review by sherifnada — APPROVED** [review]

**Pull request merged**